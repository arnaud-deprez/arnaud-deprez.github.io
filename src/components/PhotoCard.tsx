import React from 'react'
import { Card } from 'react-bootstrap'

export interface PhotoCardProps extends React.ComponentPropsWithoutRef<'div'> {
  name: string
  jobTitle: string
  // TODO: photoUrl?: string
}

export const PhotoCard = ({ name, jobTitle, ...rest }: PhotoCardProps) => (
  <Card {...rest}>
    <Card.Img
      className="rounded-circle mb-2 bg-secondary"
      style={{ height: '10rem', width: '10rem' }}
    />
    <Card.Body>
      <Card.Title className="text-center" as="h5">
        {name}
      </Card.Title>
      <Card.Subtitle className="font-italic text-muted text-center text-wrap" as="h6">
        {jobTitle}
      </Card.Subtitle>
    </Card.Body>
  </Card>
)
