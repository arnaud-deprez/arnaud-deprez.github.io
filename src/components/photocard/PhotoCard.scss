@import '../../scss/global.scss';

.photo-card.card {
  border: 0;
  background-color: inherit;
  align-items: center;

  .card-img {
    height: 10rem;
    max-width: 10rem;
  }
}