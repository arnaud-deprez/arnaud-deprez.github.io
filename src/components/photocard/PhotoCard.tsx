import React from 'react'
import { Card } from 'react-bootstrap'

import pic from '../../../static/favicon/profile.png'
import './PhotoCard.scss'

export interface PhotoCardProps extends React.ComponentPropsWithoutRef<'div'> {
  name?: string
  jobTitle?: string
  className?: string
}

export const PhotoCard = ({ name, jobTitle, className, ...rest }: PhotoCardProps) => (
  <Card className={`photo-card ${className || ''}`.trim()} {...rest}>
    <Card.Img src={pic} alt="profile" className="rounded-circle mb-2" />
    <Card.Body className="d-none d-lg-block">
      {name && (
        <Card.Title className="text-center" as="h5">
          {name}
        </Card.Title>
      )}
      {jobTitle && (
        <Card.Subtitle className="font-italic text-center text-wrap" as="h6">
          {jobTitle}
        </Card.Subtitle>
      )}
    </Card.Body>
  </Card>
)
