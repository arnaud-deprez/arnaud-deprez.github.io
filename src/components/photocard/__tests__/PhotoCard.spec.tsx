import React from 'react'
import { render } from '@testing-library/react'
import { PhotoCard } from '../PhotoCard'

describe('<PhotoCard/>', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<PhotoCard />)
    expect(asFragment()).toMatchSnapshot()
  })
  it('renders correctly with name and jobTitle', () => {
    const { asFragment } = render(<PhotoCard name="test" jobTitle="job" />)
    expect(asFragment()).toMatchSnapshot()
  })
})
