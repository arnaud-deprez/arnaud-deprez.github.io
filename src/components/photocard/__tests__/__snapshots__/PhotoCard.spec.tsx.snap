// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PhotoCard/> renders correctly 1`] = `
<DocumentFragment>
  <div
    class="photo-card card"
  >
    <img
      alt="profile"
      class="card-img rounded-circle mb-2"
      src="profile.png"
    />
    <div
      class="d-none d-lg-block card-body"
    />
  </div>
</DocumentFragment>
`;

exports[`<PhotoCard/> renders correctly with name and jobTitle 1`] = `
<DocumentFragment>
  <div
    class="photo-card card"
  >
    <img
      alt="profile"
      class="card-img rounded-circle mb-2"
      src="profile.png"
    />
    <div
      class="d-none d-lg-block card-body"
    >
      <h5
        class="text-center card-title"
      >
        test
      </h5>
      <h6
        class="font-italic text-center text-wrap card-subtitle"
      >
        job
      </h6>
    </div>
  </div>
</DocumentFragment>
`;
